src/orders/src/main/java/mushop/orders/services/OrdersService.java

```diff
@@ -2,139 +2,140 @@
-    @Autowired
-    private OrdersConfigurationProperties config;
 
-    @Autowired
-    private AsyncGetService asyncGetService;
-
-    @Autowired
-    private MessagingService messagingService;
-
-    @Autowired
+    @Inject
     private CustomerOrderRepository customerOrderRepository;
 
-    @Autowired
+    @Inject
     private MeterRegistry meterRegistry;
 
-    @Value(value = "${http.timeout:5}")
-    private long timeout;
+    @Inject
+    private OrdersPublisher ordersPublisher;
 
-    private ScheduledExecutorService cartDeleteExecutor = Executors.newScheduledThreadPool(5);
+    @Inject
+    private PaymentClient paymentClient;
 
+    @Inject
+    private AsyncGetService asyncGetService;
+
+    @Value("${http.timeout:5}")
+    private long timeout;
 
     public CustomerOrder createNewOrder(NewOrderResource orderPayload) {
         LOG.info("Creating order {}", orderPayload);
         LOG.debug("Starting calls");
         meterRegistry.counter("orders.placed").increment();
         try {
-            Future<Address> addressFuture = asyncGetService.getObject(orderPayload.address,
-                    new ParameterizedTypeReference<Address>() {
-                    });
-            Future<Customer> customerFuture = asyncGetService.getObject(orderPayload.customer,
-                    new ParameterizedTypeReference<Customer>() {
-                    });
-            Future<Card> cardFuture = asyncGetService.getObject(orderPayload.card,
-                    new ParameterizedTypeReference<Card>() {
-                    });
-            Future<List<Item>> itemsFuture = asyncGetService.getDataList(orderPayload.items,
-                    new ParameterizedTypeReference<List<Item>>() {
-                    });
-            LOG.debug("End of calls.");
-
-            //Calculate total
-            float amount = calculateTotal(itemsFuture.get(timeout, TimeUnit.SECONDS));
+            PaymentRequest paymentRequest = new PaymentRequest();
+            AtomicReference<List<Item>> orderItems = new AtomicReference<>();
+
+            boolean finished = Completable.merge(Arrays.asList(
+                    Completable.fromPublisher(asyncGetService.getObject(orderPayload.address, Address.class).doOnNext(paymentRequest::setAddress)),
+                    Completable.fromPublisher(asyncGetService.getObject(orderPayload.customer, Customer.class).doOnNext(paymentRequest::setCustomer)),
+                    Completable.fromPublisher(asyncGetService.getObject(orderPayload.card, Card.class).doOnNext(paymentRequest::setCard)),
+                    Completable.fromPublisher(asyncGetService.getDataList(orderPayload.items, Item.class).doOnNext((items -> {
+                        orderItems.set(items);
+                        //Calculate total
+                        float amount = calculateTotal(items);
+                        paymentRequest.setAmount(amount);
+                    }))))).blockingAwait(timeout, TimeUnit.SECONDS);
```

Comment:
I think all of this can be written in a manner that doesn't require calling `blocking*` methods

Reply:
Will take a look